### PR TITLE
Implement page for single news item and share functionality

### PR DESCRIPTION
**Pull request by meshnesh**
### what should this PR do?
Implement single news Item view in its own page and add functionality for sharing with other people

### How can this be manually tested?
* clone the repo
* cd into the application
* run the app to an emulator or android connected device.
* news will be displayed on the view
* click on a single news item to view it
* click the share option button and choose a medium to share through

### Relevant PT stories
* [News Detail](https://www.pivotaltracker.com/story/show/166218988)
* [Share functionality](https://www.pivotaltracker.com/story/show/166218733)

### ScreenShots
Screenshot_1558630362

Screenshot_1558630375

Screenshot_1558630392

### CODE REVIEW

File: app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java [hunk 1]
```diff
@@ -25,32 +36,40 @@
...
     private Adapter adapter;
+    private SwipeRefreshLayout swipeRefreshLayout;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+//        implement swipe down to refresh
+        swipeRefreshLayout = findViewById(R.id.swipe_refresh_layout);
```
Comment (lakivisi-zz) [inline app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java:59]: use android data binding

File: app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java [hunk 2]
```diff
@@ -105,15 +130,14 @@ public boolean onCreateOptionsMenu(Menu menu) {
         searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
             @Override
             public boolean onQueryTextSubmit(String query) {
-                if (query.length() > 2){
-                    LoadJson(query);
```
Comment (lakivisi-zz) [inline app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java]: camelcase for function names

File: app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java [hunk 3]
```diff
@@ -122,4 +146,49 @@ public boolean onQueryTextChange(String newText) {
...
+
+    private void initListener() {
+
+        adapter.setOnItemClickListener(new Adapter.OnItemClickListener() {
+            @Override
+            public void onItemClick(View view, int position) {
+                ImageView imageView = view.findViewById(R.id.img);
+                Intent intent = new Intent(MainActivity.this, NewsDetailActivity.class);
+
+                Article article = articles.get(position);
```
Comment (lakivisi-zz) [inline app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java:208]: is there a better way of accessing this data at the destination?
Or you can pass the full object and not the getters and setters then retrieve them from the destination class.

File: app/src/main/java/com/lms/mpasho_lms_news/view/details/NewsDetailActivity.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        Objects.requireNonNull(getSupportActionBar()).setTitle("");
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        final CollapsingToolbarLayout collapsingToolbarLayout = findViewById(R.id.collapsing_toolbar);
+        collapsingToolbarLayout.setTitle("");
+
+        AppBarLayout appBarLayout = findViewById(R.id.appbar);
+        appBarLayout.addOnOffsetChangedListener(this);
+        date_behavior = findViewById(R.id.date_behavior);
+        titleAppbar = findViewById(R.id.title_appbar);
```
Comment (lakivisi-zz) [inline app/src/main/java/com/lms/mpasho_lms_news/view/details/NewsDetailActivity.java:54]: Android data binding

File: app/src/main/java/com/lms/mpasho_lms_news/view/details/NewsDetailActivity.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            i.setData(Uri.parse(mUrl));
+            startActivity(i);
+            return true;
+        } else if (id == R.id.share) {
+            try {
+
+                Intent i = new Intent(Intent.ACTION_SEND);
+                i.setType("text/plan");
+                i.putExtra(Intent.EXTRA_SUBJECT, mSource);
+                String body = mTitle + "\n" + mUrl + "\n" + "Share from the News App" + "\n";
```
Comment (lakivisi-zz) [inline app/src/main/java/com/lms/mpasho_lms_news/view/details/NewsDetailActivity.java]: you can use a string builder for this

**Review by lakivisi-zz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java (hunk 1 above)
Comment (meshnesh) [reply app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java:59]: Okay, I'll implement it

File: app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java (hunk 2 above)
Comment (meshnesh) [reply app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java]: fixing it up

**Pull request merged**